### ISSUE TITLE
Support the apns-collapse-id so notifications can be coalesced

### ISSUE THREAD

**Issue by jacksonakj** [Azure/azure-notificationhubs-ios#18]
I would like to use the apns-collapse-id header to coalesce notifications on iOS devices.  However, I understand that apns-collapse-id is not supported through Azure Notification Hub based on the following post at: https://social.msdn.microsoft.com/Forums/azure/en-US/5fddec0e-1c83-48f8-8a3c-f7107899c6ba/microsoftazurenotificationhubs-version108-apnscollapseid-http2?forum=notificationhubs

**Comment by arek-avanade** [comment]
@jacksonakj I can confirm apns-collapse-id header works correctly now as far as you're using key-based authentication in Azure Notification Hubs (not certificate) for APNS.

**Comment by brannon** [comment]
That's correct. All of the `apns-x` headers should work with token (key) based APNS credentials.

Unfortunately at this time custom `apns-x` headers don't work with certificate based APNS credentials. Fixing this is on our roadmap, but there are technical limitations we need to resolve first.
